src/main/scala/com/mozilla/telemetry/mozdata/MozData.scala

```diff
@@ -0,0 +1,366 @@
+    *
+    * example:
+    *
+    * // read a little bit of raw telemetry
+    * val api = MozData(spark)
+    * val rdd = api.readRDDs(
+    *   "telemetry",
+    *   where={_.where("sourceVersion"){case "4" => true}},
+    *   fileLimit=1
+    * )
+    *
+    * @param name dataset source name
+    * @param where clauses passed to Dataset.where
+    * @param fileLimit passed to Dataset.records
+    * @param minPartitions passed to Dataset.records
+    * @return Messages read
+    */
+  def readRDD(name: String, where: Function[Dataset,Dataset] = identity,
+              fileLimit: Option[Int] = None,
+              minPartitions: Option[Int] = None): RDD[Message] = {
+    log("readRDD", Map("name" -> Some(name)))
+    where(Dataset(name)).records(fileLimit, minPartitions)(spark.sparkContext)
+  }
+
+  /** Read a table
+    *
+    * example:
+    *
+    * val api = MozData(spark)
+    *
+    * // read a global table
+    * val clientsDaily = api.readTable("clients_daily")
+    *
+    * // read v1 of nobody@mozilla.com's special_dau table
+    * val specialDauV1 = api.readTable(
+    *   tableName="special_dau",
+    *   owner=Some("nobody@mozilla.com"),
+    *   version=Some("v1"),
+    *   extraReadConfig={_.option("mergeSchema", "true")}
+    * )
+    *
+    * // read a json special_dau table defined by an s3 path
+    * val specialDauV2 = api.readTable(
+    *   tableName="special_dau",
+    *   uri=Some("s3://special-bucket/special_dau/v2"),
+    *   extraReadConfig={_.format("json")}
+    * )
+    *
+    * @param tableName table to read
+    * @param version optional specific version of table, defaults to "v1" for
+    *                new tables and the latest version for existing tables
+    * @param owner optional email that identifies non-global namespace
+    * @param uri optional non-standard location for this table
+    * @param extraReadConfig optional function to configure the DataFrameReader
+    * @return DataFrame of the requested table
+    */
+  def readTable(tableName: String, version: Option[String] = None,
+                owner: Option[String] = None, uri: Option[String] = None,
+                extraReadConfig: Function[DataFrameReader,DataFrameReader] = identity
+               ): DataFrame = {
+    val tableInfo = getTableInfo(
+      tableName=tableName,
+      version=version,
+      owner=owner,
+      uri=uri,
+      spark=spark,
+      adHocTablesDir=adHocTablesDir,
+      globalTablesDir=globalTablesDir
+    )
+
+    log(
+      action="readTable",
+      Map(
+        "detectedUri" -> tableInfo.uri,
+        "detectedVersion" -> tableInfo.version,
+        "owner" -> owner,
+        "sqlTableName" -> tableInfo.sqlTableName,
+        "tableName" -> Some(tableName),
+        "uri" -> uri,
+        "version" -> version
+      )
+    )
+
+    val reader = extraReadConfig(readConfig(spark.read))
+
+    if (tableInfo.inCatalog) {reader.table(tableInfo.sqlTableName.get)}
+    else {reader.load(tableInfo.uri.get)}
```

Comment:
nit: I think curly-braced conditionals are more readable when split across multiple lines, i.e.:
```
if (tableInfo.inCatalog) {
  reader.table(tableInfo.sqlTableName.get)
} else {
  reader.load(tableInfo.uri.get)
}
```
Automatic reformatting would fix cases like this one (Intellij's "Reformat code" or scalafmt).